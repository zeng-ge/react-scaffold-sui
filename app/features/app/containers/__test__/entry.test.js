import React from 'react'
import ConnectEntry, { Entry } from '../Entry'
import { shallow } from 'enzyme'
import rendered from 'react-test-renderer'

describe('app/containers/Entry', () => {

  beforeEach(() => {
    window.top.User = null
  })

  test('should Entry show children', () => {
    const component = shallow(
      <Entry>
        <div>entry</div>
      </Entry>
    )
    expect(component.contains(<div>entry</div>)).toBeTruthy()
  })

  test('should Entry trigger save user', () => {
    window.top.User = { name: 'sky' }
    const saveUser = jest.fn()
    shallow(
      <Entry saveUser={saveUser}  >
        <div>this is entry</div>
      </Entry>
    )
    expect(saveUser.mock.calls[0][0]).toEqual({ name: 'sky' })
  })

  test('should Entry saveUser is assigned saveUser', () => {
    window.top.User = { name: 'sky' }
    const saveUser = jest.fn()
    //包一层后saveUser不会调用
    const component = shallow(
      <div>
        <Entry saveUser={saveUser} >
          <div>this is entry</div>
        </Entry>
      </div>
    )
    const instance = component.find('Entry')
    expect(instance.prop('saveUser')).toEqual(saveUser)
  })

  test('connect Entry contains assined child', () => {
    const store = global.getStore({})
    const component = shallow(
      <ConnectEntry store={store}>
        <div>connect entry</div>
      </ConnectEntry>
    )
    expect(component.find(Entry).find('div').text()).toEqual('connect entry')
  })

  test('Entry snapshot', () => {
    const store = global.getStore({})
    const component = rendered.create(
      <ConnectEntry store={store}>
        <div>entry snapshot</div>
      </ConnectEntry>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

})
