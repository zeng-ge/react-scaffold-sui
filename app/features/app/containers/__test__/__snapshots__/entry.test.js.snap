// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/containers/Entry Entry snapshot 1`] = `
<div>
  entry snapshot
</div>
`;
